pkg/cliplugins/workload/plugin/sync.go

```diff
@@ -131,17 +132,23 @@ func enableSyncerForWorkspace(ctx context.Context, config *rest.Config, workload
 		return "", fmt.Errorf("failed to create kcp client: %w", err)
 	}
 
-	// Create the workload cluster that will serve as a point of coordination between
-	// kcp and the syncer (e.g. heartbeating from the syncer and virtual cluster urls
-	// to the syncer).
-	workloadCluster, err := kcpClient.WorkloadV1alpha1().WorkloadClusters().Create(ctx,
-		&workloadv1alpha1.WorkloadCluster{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: workloadClusterName,
-			},
-		},
-		metav1.CreateOptions{},
+	workloadCluster, err := kcpClient.WorkloadV1alpha1().WorkloadClusters().Get(ctx,
+		workloadClusterName,
+		metav1.GetOptions{},
 	)
+	if errors.IsNotFound(err) {
+		// Create the workload cluster that will serve as a point of coordination between
+		// kcp and the syncer (e.g. heartbeating from the syncer and virtual cluster urls
+		// to the syncer).
+		workloadCluster, err = kcpClient.WorkloadV1alpha1().WorkloadClusters().Create(ctx,
+			&workloadv1alpha1.WorkloadCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: workloadClusterName,
+				},
+			},
+			metav1.CreateOptions{},
+		)
+	}
 	if err != nil {
 		return "", fmt.Errorf("failed to create WorkloadCluster: %w", err)
```

Comment:
If the get call fails, and it's something other than a not-found error, the error returned here won't be correct. What if we redid the entire flow like this?

1. Try to get
2. If err != nil && !errors.IsNotFound -> return error getting
3. If NotFound
    1. Try to create
    2. If err != nil && !errors.IsAlreadyExists -> return error creating